### PR TITLE
Fix SIMD-BP128 decompressor

### PR DESCRIPTION
**Pull request by mjendruk**
There was a bug causing an integer overflow in the SIMD-BP128 decompressor, which wasn’t tested and slipped through review. I added a test and fixed the bug (and refactored the tests a bit). The real change is in `simd_bp128_decompressor.hpp:89`

Thanks @apopiak for his help.

The changes are described further in the comments.

### CODE REVIEW

File: src/lib/constant_mappings.cpp [hunk 1]
```diff
@@ -139,4 +142,16 @@ const boost::bimap<DataType, std::string> data_type_to_string =
                  return map;
                });
 
+const boost::bimap<EncodingType, std::string> encoding_type_to_string = make_bimap<EncodingType, std::string>({
+    {EncodingType::DeprecatedDictionary, "Dictionary (Deprecated)"},
```
Comment (Bouncner) [inline src/lib/constant_mappings.cpp]: Shouldn't that be gone in the current master?

Comment (mjendruk) [reply src/lib/constant_mappings.cpp]: Do you mean the deprecated dictionary column? Unfortunately not yet :(

File: src/lib/storage/vector_compression/simd_bp128/simd_bp128_decompressor.hpp [hunk 2]
```diff
@@ -69,24 +69,24 @@ class SimdBp128Decompressor : public BaseVectorDecompressor {
...
     const auto begin = _cached_meta_block_first_index;
     const auto end = _cached_meta_block_first_index + Packing::meta_block_size;
     return begin <= index && index < end;
   }
 
-  size_t _index_relative_to_cached_meta_block(size_t index) { return index - _cached_meta_block_first_index; }
+  size_t _index_relative_to_cached_meta_block(size_t index) const { return index - _cached_meta_block_first_index; }
 
-  bool _is_index_after_cached_meta_block(size_t index) {
-    return (_cached_meta_block_first_index + Packing::meta_block_size) <= index;
```
Comment (mjendruk) [inline src/lib/storage/vector_compression/simd_bp128/simd_bp128_decompressor.hpp]: `_cached_meta_block_first_index ` was initialised with `std::numeric_limits<size_t>::max()`. So the first time this method is executed, `Packing::meta_block_size` was added to the maximum value of `size_t`, which led to an integer overflow.

File: src/test/storage/encoded_column_test.cpp [hunk 3]
```diff
@@ -17,32 +21,17 @@
...
- protected:
-  static constexpr auto encoding_type_t = EncodingTypeT{};
-  static constexpr auto column_template_t = hana::at_key(encoded_column_for_type, encoding_type_t);
-
-  using ColumnTemplateType = typename decltype(column_template_t)::type;
-
-  template <typename T>
-  using EncodedColumnType = typename ColumnTemplateType::template _template<T>;
-
+class EncodedColumnTest : public BaseTestWithParam<ColumnEncodingSpec> {
```
Comment (mjendruk) [inline src/test/storage/encoded_column_test.cpp]: I changed this test from a templated to a parameterised one. Previously, `EncodingType::Dictionary` was only tested with the default `VectorCompressionType`, which is `FixedSizeByteAligned`. Now it also tests `SimdBp128`, which was buggy.

File: src/lib/constant_mappings.cpp [hunk 4]
```diff
@@ -139,4 +142,16 @@ const boost::bimap<DataType, std::string> data_type_to_string =
...
+    {EncodingType::Dictionary, "Dictionary"},
+    {EncodingType::RunLength, "RunLength"},
+});
+
+const boost::bimap<VectorCompressionType, std::string> vector_compression_type_to_string =
```
Comment (mrks) [inline src/lib/constant_mappings.cpp]: What do we need the bimap for? If, at some point, we need to convert "Fixed-size byte-aligned" (but not "fixed-size byte-aligned" or "fixed size byte aligned") to a `VectorCompressionType`, we probably did something wrong.

**Pull request merged**